src/DotNetLightning.Core/DotNetLightning.Core.fsproj

```diff
@@ -40,6 +40,12 @@
     <Compile Include="Utils/Errors.fs" />
     <Compile Include="Utils/RouteType.fs" />
     <Compile Include="Utils/PriorityQueue.fs" />
+    <Compile Include="Crypto/CryptoUtils.fs" />
```

Comment:
Why you had to move these files? I want the file order in fsproj to be sorted with directory first.

Reply:
I mean, i don't want to have something like.

```fsproj
<Compile Include="A/Foo.fs" />
<Compile Include="B/Bar.fs" />
<Compile Include="A/Buz.fs" /> <!-- A has been splitted into two! -->
```

For the sake of clarity and ease of understanding.
If this is really necessary, then we should introduce another repository.

Reply:
It's no clear to me what the best way to break the files into separate dirs would be. Aezeed is closely related to keys, but it depends on the serialization types (`BitWriter` and `LightningWriterStream`), these types (specifically `LightningReaderStream`) depend on the key types though.

Perhaps the serialization methods should be moved from `LightningReaderStream` onto the key types as an interface? eg. have the key types implement the `ILightningSerializable` interface, then make `Lightning{Reader,Writer}Stream` be able to read/write any `ILightningSerializable` type?

Reply:
Why don't you just put new key types in `Utils/` and extend its methods in `Crypto/` ? I think that is more easier to reason about which logic is in which folder.

Reply:
To rephrase my thinking for the sake of clarity....

*The entire `Serialization/` folder should work without any cryptographic stuff. So putting `Crypto/` above `Serialization/` must be unnecessary and responsibility gets less segregated.
* Why you want to do this is because (I guess) you want to use new Key types as both 1. Primitive building block for serialization (like LNMoney) and 2. entrypoint for cryptographic operation (e.g. key tweaking). But these two characteristics are logically in different layer. So crypto must come after `Serialization/` and if you want new key types to have cryptographic methods, it must be in the form of extension methods in `Crypto/`

Reply:
> Why don't you just put new key types in `Utils/` and extend its methods in `Crypto/` ?

Because I didn't think of that :)

I've done that now.